src/main/java/cubicchunks/converter/headless/command/commands/ConverterNameCommand.java

```diff
@@ -26,21 +26,26 @@
 import com.mojang.brigadier.CommandDispatcher;
 import com.mojang.brigadier.builder.LiteralArgumentBuilder;
 import cubicchunks.converter.headless.command.HeadlessCommandContext;
+import cubicchunks.converter.lib.Registry;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 public class ConverterNameCommand {
     public static void register(CommandDispatcher<HeadlessCommandContext> dispatcher) {
-        dispatcher.register(LiteralArgumentBuilder.<HeadlessCommandContext>literal("converter")
-            .then(LiteralArgumentBuilder.<HeadlessCommandContext>literal("Default")
-                .executes((context) -> {
-                    context.getSource().setConverterName("Default");
-                    return 1;
-                })
-            ).then(LiteralArgumentBuilder.<HeadlessCommandContext>literal("Relocating")
-                .executes((context) -> {
-                    context.getSource().setConverterName("Relocating");
-                    return 1;
-                })
-            )
-        );
+        dispatcher.register(StreamSupport.stream(Registry.getConverters().spliterator(), false)
+                .map(Registry.ClassTriple::getConverter)
+                .map(Registry::getConverterId)
+                .distinct()
+                .reduce(LiteralArgumentBuilder.literal("converter"),
+                        (builder, name) -> builder.then(LiteralArgumentBuilder.<HeadlessCommandContext>literal(name)
+                                .executes(context -> {
+                                    context.getSource().setConverterName(name);
+                                    return 1;
+                                })),
+                        (a, b) -> {
+                            throw new UnsupportedOperationException();
```

Comment:
Is this never getting called actually guaranteed by documentation? To my knowledge, it's never specified explicitly, just a vague implication from specific streams being sequential.

Reply:
no, it isn't, although i see no reason for any sane implementation to use the `combiner` if the stream isn't parallel. i'll work around that if you prefer, though (i just liked being able to do the entire thing in a single statement, rather than manually updating a builder variable in a loop).

Reply:
It's fine, I was just wondering if there is maybe something I missed in my interpretation, as I wanted to use reduce() that way before but never did because of this being unspecified